Another nearly clear run on M1: just this failure, which seems to me to be a brittle test. The output is the same as the expected, except for the order of parameters:

```
[Check failure on line 19 in integration-tests/smallrye-config/src/test/java/io/quarkus/it/smallrye/config/AppConfigTest.java](https://github.com/quarkusio/quarkus/commit/b06790ddd8dd7357595cb5b9785790a1277d1d3d#annotation_4644614958)

@quarkus-bot quarkus-bot / Build summary for b06790ddd8dd7357595cb5b9785790a1277d1d3d

JVM Tests - JDK 17 MacOS M1

java.lang.AssertionError: 
1 expectation failed.
Response body doesn't match expectation.

java.lang.AssertionError: 
1 expectation failed.
Response body doesn't match expectation.
Expected: is "AppConfig{name=app, info=Info{count=10, alias=alias}}"
  Actual: AppConfig{name=app, info=Info{alias=alias, count=10}}

	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
```

--- a/.github/filter-jvm-tests-json.sh
+++ b/.github/filter-jvm-tests-json.sh
@@ -1,0 +1,20 @@
+#!/bin/bash
+
+BASEDIR=$(dirname "$0")
+
+# Filter out mac os from a matrix of build configurations.
+# The reason we do this is that running mac on a fork won't work; the fork won't have a self-hosted runner
+
+# See https://stackoverflow.com/questions/65384420/how-to-make-a-github-action-matrix-element-conditional
+
+repoName=${GITHUB_REPOSITORY}
+
+if [[ $repoName == "quarkusio/quarkus" ]]
+then
+    matrix=$(cat $BASEDIR/matrix-jvm-tests.json)
+else
+  # Use jq to read in a json file that represents the matrix configuration.
+  matrix=$(jq 'map(. | select(.["os-name"]!="macos-arm64-latest"))' $BASEDIR/matrix-jvm-tests.json)
+fi
+
+echo \{java: ${matrix}\}

--- a/.github/matrix-jvm-tests.json
+++ b/.github/matrix-jvm-tests.json
@@ -1,0 +1,36 @@
+[ {
+  "name": "11",
+  "java-version": 11,
+  "maven_args": "$JVM_TEST_MAVEN_ARGS",
+  "maven_opts": "-Xmx2g -XX:MaxMetaspaceSize=1g",
+  "os-name": "ubuntu-latest"
+}
+, {
+  "name": "17",
+  "java-version": 17,
+  "maven_args": "$JVM_TEST_MAVEN_ARGS",
+  "maven_opts": "-Xmx2g -XX:MaxMetaspaceSize=1g",
+  "os-name": "ubuntu-latest"
+}
+, {
+  "name": "18",
+  "java-version": 18,
+  "maven_args": "$JVM_TEST_MAVEN_ARGS",
+  "maven_opts": "-Xmx2g -XX:MaxMetaspaceSize=1g",
+  "os-name": "ubuntu-latest"
+}
+, {
+  "name": "11 Windows",
+  "java-version": 11,
+  "maven_args": "-DskipDocs -Dformat.skip",
+  "maven_opts": "-Xmx2g -XX:MaxMetaspaceSize=1g -Xlog:gc*=debug:file=windows-java-11.txt",
+ "os-name": "windows-latest"
+}
+, {
+  "name": "17 MacOS M1",
+  "java-version": 17,
+  "maven_args": "-DskipDocs -Dformat.skip",
+  "maven_opts": "-Xmx2g -XX:MaxMetaspaceSize=1g",
+  "architecture": "aarch64",
+  "os-name": "macos-arm64-latest"
+}]

--- a/.github/workflows/ci-actions-incremental.yml
+++ b/.github/workflows/ci-actions-incremental.yml
@@ -222,6 +222,7 @@ jobs:
       GIB_IMPACTED_MODULES: ${{ needs.build-jdk11.outputs.gib_impacted }}
     outputs:
       native_matrix: ${{ steps.calc-native-matrix.outputs.matrix }}
+      jvm_matrix: ${{ steps.calc-jvm-matrix.outputs.matrix }}
       run_jvm: ${{ steps.calc-run-flags.outputs.run_jvm }}
       run_devtools: ${{ steps.calc-run-flags.outputs.run_devtools }}
       run_gradle: ${{ steps.calc-run-flags.outputs.run_gradle }}
@@ -234,6 +235,12 @@ jobs:
         run: |
           echo "GIB_IMPACTED_MODULES: ${GIB_IMPACTED_MODULES}"
           json=$(.github/filter-native-tests-json.sh "${GIB_IMPACTED_MODULES}" | tr -d '\n')
+          echo "${json}"
+          echo "::set-output name=matrix::${json}"
+      - name: Calculate matrix from matrix-jvm-tests.json
+        id: calc-jvm-matrix
+        run: |
+          json=$(.github/filter-jvm-tests-json.sh)
           echo "${json}"
           echo "::set-output name=matrix::${json}"
       - name: Calculate run flags
@@ -267,40 +274,12 @@ jobs:
       MAVEN_OPTS: ${{ matrix.java.maven_opts }}
     strategy:
       fail-fast: false
-      matrix:
-        java:
-          - {
-            name: "11",
-            java-version: 11,
-            maven_args: "$JVM_TEST_MAVEN_ARGS",
-            maven_opts: "-Xmx2g -XX:MaxMetaspaceSize=1g",
-            os-name: "ubuntu-latest"
-          }
-          - {
-            name: "17",
-            java-version: 17,
-            maven_args: "$JVM_TEST_MAVEN_ARGS",
-            maven_opts: "-Xmx2g -XX:MaxMetaspaceSize=1g",
-            os-name: "ubuntu-latest"
-          }
-          - {
-            name: "18",
-            java-version: 18,
-            maven_args: "$JVM_TEST_MAVEN_ARGS",
-            maven_opts: "-Xmx2g -XX:MaxMetaspaceSize=1g",
-            os-name: "ubuntu-latest"
-          }
-          - {
-            name: "11 Windows",
-            java-version: 11,
-            maven_args: "-DskipDocs -Dformat.skip",
-            maven_opts: "-Xmx2g -XX:MaxMetaspaceSize=1g -Xlog:gc*=debug:file=windows-java-11.txt",
-            os-name: "windows-latest"
-          }
+      matrix: ${{ fromJson(needs.calculate-test-jobs.outputs.jvm_matrix) }}
+
 
     steps:
       - name: Stop mysql
-        if: "!startsWith(matrix.java.os-name, 'windows')"
+        if: "!startsWith(matrix.java.os-name, 'windows') && !startsWith(matrix.java.os-name, 'macos')"
         shell: bash
         run: |
           ss -ln
@@ -316,12 +295,12 @@ jobs:
         run: git remote add quarkusio https://github.com/quarkusio/quarkus.git
 
       - name: apt clean
-        if: "!startsWith(matrix.java.os-name, 'windows')"
+        if: "!startsWith(matrix.java.os-name, 'windows') && !startsWith(matrix.java.os-name, 'macos')"
         shell: bash
         run: sudo apt-get clean
 
       - name: Reclaim Disk Space
-        if: "!startsWith(matrix.java.os-name, 'windows')"
+        if: "!startsWith(matrix.java.os-name, 'windows') && !startsWith(matrix.java.os-name, 'macos')"
         run: .github/ci-prerequisites.sh
 
       - name: Set up JDK ${{ matrix.java.java-version }}
@@ -329,6 +308,7 @@ jobs:
         with:
           distribution: temurin
           java-version: ${{ matrix.java.java-version }}
+          architecture: ${{ matrix.java.architecture || 'x64' }}
 
       - name: Download Maven Repo
         uses: actions/download-artifact@v3

--- a/extensions/smallrye-graphql/deployment/pom.xml
+++ b/extensions/smallrye-graphql/deployment/pom.xml
@@ -122,4 +122,31 @@
         </plugins>
     </build>
 
+    <profiles>
+        <!-- Disabled pending fix of #28033 -->
+        <profile>
+            <id>mac-m1</id>
+            <activation>
+                <os>
+                    <family>mac</family>
+                    <arch>aarch64</arch>
+                </os>
+                <property>
+                    <name>env.GITHUB_ACTIONS</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
 </project>

--- a/integration-tests/hibernate-orm-graphql-panache/pom.xml
+++ b/integration-tests/hibernate-orm-graphql-panache/pom.xml
@@ -119,4 +119,31 @@
         </plugins>
     </build>
 
+    <profiles>
+        <!-- Disabled pending fix of #28035 -->
+        <profile>
+            <id>mac-m1</id>
+            <activation>
+                <os>
+                    <family>mac</family>
+                    <arch>aarch64</arch>
+                </os>
+                <property>
+                    <name>env.GITHUB_ACTIONS</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
 </project>

--- a/integration-tests/hibernate-orm-rest-data-panache/pom.xml
+++ b/integration-tests/hibernate-orm-rest-data-panache/pom.xml
@@ -139,4 +139,30 @@
         </plugins>
     </build>
 
+    <profiles>
+        <!-- Disabled pending fix of #28035 -->
+        <profile>
+            <id>mac-m1</id>
+            <activation>
+                <os>
+                    <family>mac</family>
+                    <arch>aarch64</arch>
+                </os>
+                <property>
+                    <name>env.GITHUB_ACTIONS</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/integration-tests/jackson/pom.xml
+++ b/integration-tests/jackson/pom.xml
@@ -85,4 +85,30 @@
         </plugins>
     </build>
 
+    <profiles>
+        <!-- Disabled pending fix of #28035 -->
+        <profile>
+            <id>mac-m1</id>
+            <activation>
+                <os>
+                    <family>mac</family>
+                    <arch>aarch64</arch>
+                </os>
+                <property>
+                    <name>env.GITHUB_ACTIONS</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/integration-tests/spring-data-rest/pom.xml
+++ b/integration-tests/spring-data-rest/pom.xml
@@ -139,4 +139,30 @@
         </plugins>
     </build>
 
+    <profiles>
+        <!-- Disabled pending fix of #28035 -->
+        <profile>
+            <id>mac-m1</id>
+            <activation>
+                <os>
+                    <family>mac</family>
+                    <arch>aarch64</arch>
+                </os>
+                <property>
+                    <name>env.GITHUB_ACTIONS</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

